pkg/cache/clusterqueue.go

```diff
@@ -196,6 +198,17 @@ func (c *clusterQueue) updateQuotasAndResourceGroups(in []kueue.ResourceGroup) b
 }
 
 func (c *clusterQueue) updateQueueStatus() {
+	if features.Enabled(features.TopologyAwareScheduling) &&
+		len(c.tasFlavors) > 0 &&
+		len(c.workloadsNotAccountedForTAS) > 0 &&
+		c.isTASSynced() {
```

Comment:
Do we need to return if we are not synced to avoid setting status to active?

Reply:
If `c.isTASSynced()` returns false, then `isTASViolated` will return true, and the CQ will not be considered active. In that case the scheduling is blocked, because the snapshot does not build, see here: https://github.com/kubernetes-sigs/kueue/blob/f9e897998e54b5cf69f9318606eeffc1a75640a3/pkg/cache/snapshot.go#L136-L139

Yes, I know it is tricky, but I think it will work ok.